lib/plugins/cloud-watch.js

```diff
@@ -16,21 +21,54 @@ function register(skill, cloudwatch, eventMetric) {
     Namespace: 'StateMachine',
   };
 
-  const eventMetricConfig = _.merge(defaulEventMetric, eventMetric);
+  if (config.MetricName) {
+    _.set(defaulEventMetric, 'MetricData[0].MetricName', config.MetricName);
+  }
+
+  if (config.Namespace) {
+    _.set(defaulEventMetric, 'Namespace', config.Namespace);
+  }
+
+  const onErrorMetric = _.cloneDeep(defaulEventMetric);
+  _.set(onErrorMetric, 'MetricData[0].Value', 1);
 
-  const putMetricDataAsync = Promise.promisify(cloudwatch.putMetricData);
-  skill.onStateMachineError(alexaEvent => Promise.try(() => putMetricDataAsync(eventMetricConfig)
-  .then(() => {
+  const OnSuccessMetric = _.cloneDeep(defaulEventMetric);
+  _.set(OnSuccessMetric, 'MetricData[0].Value', 0);
+
+  const putMetricDataAsync = Promise.promisify(cloudwatch.putMetricData.bind(cloudwatch));
+
+  skill.onBeforeReplySent(alexaEvent => Promise.try(() => {
+    _.set(OnSuccessMetric, 'MetricData[0].Timestamp', new Date());
+    return putMetricDataAsync(OnSuccessMetric).then((data, error) => {
+      debug(`onBeforeReplySent Metric Data Error: ${JSON.stringify(error)}`);
+      debug(`onBeforeReplySent Metric Data Success: ${JSON.stringify(data)}`);
+      return null;
+    });
+  }));
+
+  skill.onStateMachineError(alexaEvent => Promise.try(() => {
     _.set(alexaEvent, 'cloudwatch.errorReported', true);
-    return null;
-  })));
+    _.set(onErrorMetric, 'MetricData[0].Timestamp', new Date());
 
+    return putMetricDataAsync(onErrorMetric).then((data, error) => {
+      debug(`onStateMachineError Metric Data Error: ${JSON.stringify(error)}`);
```

Comment:
The intention is that in case that the metric data couldn't be sent, tell the developer whats wrong.